test/getRecipeById.test.js

```diff
@@ -41,35 +43,70 @@ const res = { json, data: null };
 
 describe('getRecipeById testing', () => {
     it('should fail if no id in request', () => {
-        const req = { params: { id: '' } };
+        const { getRecipeById: proxiedgetRecipeById } = proxyquire(
+            '../controllers/userController', {
+                './signinController': {
+                    sessions: { 'coo coo': null },
+                },
+            },
+        );
+        const req = { params: { id: '' } , cookies: { access: 'coo coo' } };
 
         getRecipeById(req, res);
         expect(res.data).to.equal('Wrong id');
     });
 
     it('should fail if id is not numeric string', () => {
-        const req = { params: { id: 'q' } };
+        const { getRecipeById: proxiedgetRecipeById } = proxyquire(
+            '../controllers/userController', {
+                './signinController': {
+                    sessions: { 'coo coo': null },
+                },
+            },
+        );
+        const req = { params: { id: 'q' } , cookies: { access: 'coo coo' } };
 
         getRecipeById(req, res);
         expect(res.data).to.equal('Wrong id');
     });
 
     it('should fail if id is =0', () => {
-        const req = { params: { id: '0' } };
+        const { getRecipeById: proxiedgetRecipeById } = proxyquire(
+            '../controllers/userController', {
+                './signinController': {
+                    sessions: { 'coo coo': null },
+                },
+            },
+        );
+        const req = { params: { id: '0' } , cookies: { access: 'coo coo' } };
```

Comment:
why id is a string?

Reply:
we have in controller method getRecipeById in which we have (!id.match(/^[0-9]+$/) condition where checking  id with string method